spock-core/src/main/java/org/spockframework/runtime/GlobalExtensionRegistry.java

```diff
@@ -82,26 +80,61 @@ public Object getConfigurationByName(String name) {
   private void verifyGlobalExtension(Class<?> clazz) {
     if (!IGlobalExtension.class.isAssignableFrom(clazz))
       throw new ExtensionException(
-          "Class '%s' is not a valid global extension because it is not derived from '%s'"
+        "Class '%s' is not a valid global extension because it is not derived from '%s'"
       ).withArgs(clazz.getName(), IGlobalExtension.class.getName());
   }
 
-  private IGlobalExtension instantiateGlobalExtension(Class<?> clazz) {
+  private IGlobalExtension instantiateAndConfigureGlobalExtension(Class<? extends IGlobalExtension> clazz) {
```

Comment:
Part of my suggestion was, that this method is now superfluous, it is just an unnecessary "rename" of the following method, just use `instantiateAndConfigureExtension` directly. :-)